ReactWindows/ReactNative.Shared/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,229 @@
+                                                                                                typeof(UIElement),
+                                                                                                new PropertyMetadata(ImportantForAccessibility.Auto));
+
+        /// <summary>
+        /// ImportantForAccessibilityAttached property setter.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <param name="value"></param>
+        public static void SetImportantForAccessibilityAttached(UIElement element, ImportantForAccessibility value)
+        {
+            element.SetValue(ImportantForAccessibilityAttachedProperty, value);
+        }
+
+        /// <summary>
+        /// ImportantForAccessibilityAttached property getter.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <returns></returns>
+        public static ImportantForAccessibility GetImportantForAccessibilityAttached(UIElement element)
+        {
+            return (ImportantForAccessibility)element.GetValue(ImportantForAccessibilityAttachedProperty);
+        }
+
+        /// <summary>
+        /// Recursively sets AccessibilityView to accessibilityView for parentAutomationPeer children.
+        /// </summary>
+        /// <param name="parentAutomationPeer"></param>
+        /// <param name="accessibilityView"></param>
+        private static void SetChildrenAccessibilityView(AutomationPeer parentAutomationPeer, AccessibilityView accessibilityView)
+        {
+            if (parentAutomationPeer?.GetChildren() == null)
+            {
+                return;
+            }
+
+            foreach (AutomationPeer childAutomationPeer in parentAutomationPeer.GetChildren())
+            {
+                if (childAutomationPeer is FrameworkElementAutomationPeer childFEAutomationPeer)
+                {
+                    if (childFEAutomationPeer?.Owner != null)
+                    {
+                        AutomationProperties.SetAccessibilityView(childFEAutomationPeer.Owner, accessibilityView);
+                    }
+                }
+                else if (childAutomationPeer is ItemAutomationPeer childIAutomationPeer)
+                {
+                    if (childIAutomationPeer?.Item is UIElement childUIElement)
+                    {
+                        AutomationProperties.SetAccessibilityView(childUIElement, accessibilityView);
+                    }
+                }
+
+                SetChildrenAccessibilityView(childAutomationPeer, accessibilityView);
+            }
+        }
+
+        /// <summary>
+        /// Returns true if in uiElementAutomationPeer parents chain there is an UIElement with
+        /// ImportantForAccessibility value of Yes or NoHideDescendants.
+        /// </summary>
+        /// <param name="uiElementAutomationPeer"></param>
+        /// <returns></returns>
+        private static bool IsHiddenElement(AutomationPeer uiElementAutomationPeer)
```

Comment:
>IsHiddenElement [](start = 28, length = 15)

Is there a clearer method name we could use here? Based on the method name, I'm surprised that this method could return false if `uiElementAutomationPeer` has `importantForAccessibility` set to `No` or `NoHideDescedants`.

Perhaps a clearer name would be `IsHiddenByAncestor`. Additionally, we might want to have a comment about the `Yes` case because it seems counter intuitive that `Yes` would hide the descendants.

Reply:
Agree that IsHiddenByAncestor conducts more meaning so changing it and adding some comments.

---
In reply to: [163340596](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163340596) [](ancestors = 163340596)